// dao/mysql.js
/**
 * Created by john on 2017/4/16.
 */
var mysql=require('mysql');
//连接数据库
function connectServer(){
    var client=mysql.createConnection({
        host:'localhost',
        user:'root',
        password:'12345',
        database:'test'
    });
    return client;
}
//在user表中查找某用户名的密码
function  selectUsr(client,username,callback){
    //client为一个mysql连接对象
    client.query('select password from user where username="'+username+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//在user表中插入一行用户名和密码

function insertUsr(client , username , password,callback){
    client.query('insert into user value(?,?)', [username, password], function(err,result){
        if( err ){
            console.log( "error:" + err.message);
            return err;
        }
        callback(err);
    });
}
//在matches表中查找某用户名的全部画布id
function  selectMts(client,username,callback){
    //client为一个mysql连接对象
    client.query('select canvasid from matches where username="'+username+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//在matches表中某用户名是否连接某一画布
function ifmatch(client,username,canvasid,callback){
    //client为一个mysql连接对象
    client.query('select * from matches where username="'+username+'"and canvasid="'+canvasid+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//在画布表中查找某一画布id的内容
function  selectOpt(client,canvasid,callback){
    //client为一个mysql连接对象
    client.query("select * from cooperation where canvasid='"+canvasid+"'",function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//在matches表中插入一行用户名和画布的连接
function insertMts(client , username , canvasid,callback){
    client.query('insert into matches value(?,?)', [username, canvasid], function(err,result){
        if( err ){
            console.log( "error:" + err.message);
            return err;
        }
        callback(err);
    });
}
//在画布表中插入新画布
function insertCav(client , username,callback){
    client.query('insert into cooperation value(?,?,?,?,?,?,?,?,?,?,?,?)', [username,,'未命名','重要伙伴','关键业务','核心资源','价值主张','客户关系','渠道通路','客户细分','成本结构','收入结构'], function(err,resu){
        if( err ){
            console.log( "error:" + err.message);
            return err;
        }
        callback(err);
    });
}
//选择画布表中新加入的一行画布内容
function selectlast(client ,callback){
    client.query( 'select * from cooperation order by canvasid desc limit 0,1',function (err,result,field) {
            if(err) throw err;
            callback(result);
    });
}
//在画布表中查找是否有某一用户编辑的某一画布
function  selectCava(client,username,canvasid,callback){
    //client为一个mysql连接对象
    client.query('select * from cooperation where canvasid="'+canvasid+'"and username="'+username+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//删除某一画布以及他对所有用户名的连接

function delCava(client,username,canvasid,callback) {
    client.query('delete from cooperation where canvasid="'+canvasid+'"and username="'+username+'"',function(err,results,fields){
        if(err) throw err;
    });
    client.query('delete from matches where canvasid="'+canvasid+'"',function(err,results,fields){
        if(err) throw err;
        callback(err);
    });
}
//在matches表中删除某一画布和某一用户的连接
function delMts(client,username,canvasid,callback) {
    client.query('delete from matches where canvasid="'+canvasid+'"and username="'+username+'"',function(err,results,fields){
        if(err) throw err;
        callback(err);
    });
}

exports.connect = connectServer;
exports.selectUsr  = selectUsr;
exports.insertUsr = insertUsr;
exports.selectMts = selectMts;
exports.selectOpt = selectOpt;
exports.insertMts = insertMts;
exports.insertCav = insertCav;
exports.selectlast= selectlast;
exports.ifmatch= ifmatch;
exports.selectCava= selectCava;
exports.delCava= delCava;
exports.delMts= delMts;
